### PR TITLE
COMMERCE-12514 - Implement PayPal Payment Integration and replace old PayPal Payment Method for standard payments

### PR DESCRIPTION
**Pull request by ghost**
https://liferay.atlassian.net/browse/COMMERCE-12514

### CODE REVIEW

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce-functional

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:commerce-functional

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-payment-service/src/main/java/com/liferay/commerce/payment/internal/gateway/CommercePaymentGatewayImpl.java [hunk 1]
```diff
@@ -94,6 +111,20 @@ public CommercePaymentEntry authorize(
...
 		}
 
+		if (StringUtil.equals(
+				commercePaymentEntry.getClassName(),
+				CommerceOrder.class.getName())) {
+
+			PermissionThreadLocal.setPermissionChecker(
+				PermissionCheckerFactoryUtil.create(currentUser));
+
+			updateOrderPaymentStatus(
```
Comment (luca-pellizzon) [inline modules/apps/commerce/commerce-payment-service/src/main/java/com/liferay/commerce/payment/internal/gateway/CommercePaymentGatewayImpl.java:121]: I would decouple this from the PaymentGateway. I would create a messageListener on the CommercePaymentEntry status change, that will change the order paymentStatus if we are handling a CommerceOrder.

**Review by luca-pellizzon — CHANGES_REQUESTED** [review]

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by luca-pellizzon** [comment]
https://github.com/liferay-commerce/liferay-portal/pull/4403

**Pull request closed without merging**